drivers/gpio/gpio_mcp23xxx.c

```diff
@@ -434,7 +434,10 @@ static void mcp23xxx_work_handler(struct k_work *work)
 	intf &= level_ints | (intcap & drv_data->rising_edge_ints) |
 		(~intcap & drv_data->falling_edge_ints);
 
+	k_sem_give(&drv_data->lock);
 	gpio_fire_callbacks(&drv_data->callbacks, dev, intf);
+	return;
+
 done:
```

Comment:
Perhaps rename the label from `done` to `unlock` or `failed` or similar, as this is not really "done" now? Otherwise looks good.

Reply:
Thanks, changed to `fail`.